src/TwinGet.AutomationInterface/AutomationInterface.cs

```diff
@@ -0,0 +1,100 @@
+﻿// This file is licensed to you under MIT license.
+
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using TwinGet.AutomationInterface.ComMessageFilter;
+using TwinGet.AutomationInterface.Exceptions;
+using TwinGet.AutomationInterface.Utils;
+
+namespace TwinGet.AutomationInterface
+{
+    [SupportedOSPlatform("windows")]
+    public class AutomationInterface : IDisposable
+    {
+        public string ProgId { get => _progId; }
+        private string _progId;
+        private readonly EnvDTE80.DTE2? _dte;
+        private bool _disposedValue;
+
+        public AutomationInterface()
+        {
+            MessageFilter.Register();
+            _dte = TryInitializeDte(out _progId);
+            if (_dte is null) { throw new CouldNotCreateTwinCatDte("Is TwinCAT installed in this system?"); }

```

Comment:
Depending on how this message is used, would be better to give more context, that the DTE creation failed, and as an additive tip to make sure TwinCAT is installed.